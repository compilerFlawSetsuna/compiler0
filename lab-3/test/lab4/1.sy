int a;

int main()
{
    int a;
    a = 1 + 2;
   if(a < 0x5)
        return 1;
    return 0;
	int b=017;
	int c=0x1a;
}
/* ??? */
