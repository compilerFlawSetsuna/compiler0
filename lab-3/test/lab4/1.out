     token    lexeme    lineno       val
       INT       int         1         0
        ID         a         1         1
 SEMICOLON         ;         1         0
       INT       int         3         0
        ID      main         3         1
    LPAREN         (         3         0
    RPAREN         )         3         0
    LBRACE         {         4         0
       INT       int         5         0
        ID         a         5         4
 SEMICOLON         ;         5         0
        ID         a         6         4
    ASSIGN         =         6         0
       INT         1         6         1
       ADD         +         6         0
       INT         2         6         2
 SEMICOLON         ;         6         0
        IF        if         7         0
    LPAREN         (         7         0
        ID         a         7         4
      LESS         <         7         0
       INT       0x5         7         5
    RPAREN         )         7         0
    RETURN    return         8         0
       INT         1         8         1
 SEMICOLON         ;         8         0
    RETURN    return         9         0
       INT         0         9         0
 SEMICOLON         ;         9         0
       INT       int        10         0
        ID         b        10         4
    ASSIGN         =        10         0
       INT       017        10        15
 SEMICOLON         ;        10         0
       INT       int        11         0
        ID         c        11         4
    ASSIGN         =        11         0
       INT      0x1a        11        26
 SEMICOLON         ;        11         0
    RBRACE         }        12         0
